scripts/shared/lib/deploy_bundle

```diff
@@ -0,0 +1,249 @@
+      echo "[ERROR] CatalogSource is not ready."
+      exit 1
+  fi
+
+  # Debug
+  kubectl -n ${MARKETPLACE_NAMESPACE} get catalogsource --ignore-not-found
+  kubectl -n ${MARKETPLACE_NAMESPACE} get pods --ignore-not-found
+  kubectl -n ${MARKETPLACE_NAMESPACE} get packagemanifests --ignore-not-found | grep 'Submariner Catalog Source' || true
+
+
+  # Create the OperatorGroup
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: submariner-group
+  namespace: ${SUBM_NS}
+spec:
+  targetNamespaces:
+    - ${SUBM_NS}
+EOF
+
+  # Debug
+  kubectl get og -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Catalog source created"
+}
+
+function install_bundle() {
+  local installPlan
+
+  # Delete previous catalogSource and Subscription
+  kubectl delete sub/submariner-subscription -n "${SUBM_NS}" --wait --ignore-not-found
+
+  # Create the Subscription (Approval should be Manual not Automatic in order to pin the bundle version)
+  echo "[INFO](${cluster}) Install the bundle..."
+cat <<EOF | kubectl apply -f -
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: submariner-subscription
+  namespace: ${SUBM_NS}
+spec:
+  channel: ${CHANNEL}
+  installPlanApproval: Manual
+  name: ${OPERATOR_NAME}
+  source: submariner-catalog-source
+  sourceNamespace: ${MARKETPLACE_NAMESPACE}
+  startingCSV: ${OPERATOR_NAME}.v${VERSION}
+EOF
+
+  # Manual Approve
+  echo "[INFO](${cluster}) Approve the installPlan..."
+  kubectl wait --for condition=InstallPlanPending --timeout=5m -n "${SUBM_NS}" subs/submariner-subscription || (echo "[ERROR](${cluster}) InstallPlan not found."; exit 1)
+  installPlan=$(kubectl get subscriptions.operators.coreos.com submariner-subscription -n "${SUBM_NS}" -o jsonpath='{.status.installPlanRef.name}')
+  if [ -n "${installPlan}" ]; then
+    kubectl patch installplan -n "${SUBM_NS}" "${installPlan}" -p '{"spec":{"approved":true}}' --type merge
+  fi
+
+  # Debug
+  kubectl get sub -n "${SUBM_NS}" --ignore-not-found
+  kubectl get installplan -n "${SUBM_NS}" --ignore-not-found
+  kubectl get csv -n "${SUBM_NS}" --ignore-not-found
+  kubectl get pods -n "${SUBM_NS}" --ignore-not-found
+
+  echo "[INFO](${cluster}) Bundle installed"
+}
+
+function setup_broker() {
+  local brokerClientSecret
+
+  if ! (timeout 5m bash -c "until kubectl --context=${cluster} get crds brokers.submariner.io > /dev/null 2>&1; do sleep 10; done"); then
+      echo "[ERROR] Broker CRD was not found."
+      exit 1
+  fi
+
+  # Create the broker Namespace & RBAC
+  echo "[INFO] Deploy the broker"
+  [ -d /tmp/submariner-operator ] && rm -Rf /tmp/submariner-operator
+  git clone --depth 1 --single-branch --branch "${BASE_BRANCH}" https://github.com/submariner-io/submariner-operator /tmp/submariner-operator
```

Comment:
This makes me somewhat uneasy, but I’m not sure it’s really any different than using the `devel` `subctl` when deploying using the latter...

Reply:
@skitt yes, there are some difference. `subctl` handles everything around the broker deployment whether it's RBAC or SCC. We would like to make sure we don't miss anything in our guide for users who are not using subctl.

Reply:
I'd clone only if it doesn't exist, as for operator you may want to test unmerged code and cloning from github would prohibit that.

Reply:
@dfarrell07 @mkolesnik you are right. The idea was to run this test periodically but it can definitely be part of the PR tests.

Reply:
> @skitt yes, there are some differences. `subctl` handles everything around the broker deployment whether it's RBAC or SCC. We would like to make sure we don't miss anything in our guide for users who are not using subctl.

Sorry, my comment wasn’t clear — I wasn’t commenting on differences between `subctl` and the bundle deployment, but rather the fact that we’re cloning the repo from the devel branch, which introduces a similar dependency as using devel `subctl`.

Reply:
@skitt thanks for your comment, so after the last change and as @mkolesnik was suggesting, I clone the repo only if it does not exist.